lib/ntlmauth/ntlmauth.h

```diff
@@ -32,24 +32,25 @@ extern "C" {
 #define NTLM_REQUEST_NON_NT_SESSION_KEY     0x400000
 
 /* NTLM error codes */
-#define NTLM_ERR_INTERNAL         -3
-#define NTLM_ERR_BLOB             -2
-#define NTLM_ERR_BAD_PROTOCOL     -1
-#define NTLM_ERR_NONE              0    /* aka. SMBLM_ERR_NONE */
-/* codes used by smb_lm helper */
-#define NTLM_ERR_SERVER            1    /* aka. SMBLM_ERR_SERVER   */
-#define NTLM_ERR_PROTOCOL          2    /* aka. SMBLM_ERR_PROTOCOL */
-#define NTLM_ERR_LOGON             3    /* aka. SMBLM_ERR_LOGON    */
-#define NTLM_ERR_UNTRUSTED_DOMAIN  4
-#define NTLM_ERR_NOT_CONNECTED     10
-/* codes used by mswin_ntlmsspi helper */
-#define NTLM_SSPI_ERROR         1
-#define NTLM_BAD_NTGROUP        2
-#define NTLM_BAD_REQUEST        3
-/* TODO: reduce the above codes down to one set non-overlapping. */
+enum class NtlmError : unsigned char
```

Comment:
I suggest the following change to address the signed/unsigned/type conversion problems discussed in other change requests:

```suggestion
enum class NtlmError : int
```

Using `int` might also be faster, but that is not why I am suggesting this.

Reply:
I think we can just not force the type then, and let the compiler do its magic. I believe that int is the default